src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -996,6 +1020,74 @@ pub mod tests {
         assert_eq!(net.acked_features, features);
     }
 
+    #[test]
+    // We can't get offload features, that were set up to a TAP device by ioctl,
+    // hence that we have to validate, that we sort out unsupported features correctly
+    fn test_build_tap_supported_features_all() {
+        let desired_features = [
+            (gen::TUN_F_CSUM, VIRTIO_NET_F_CSUM),
+            (gen::TUN_F_UFO, VIRTIO_NET_F_GUEST_UFO),
+            (gen::TUN_F_TSO4, VIRTIO_NET_F_GUEST_TSO4),
+            (gen::TUN_F_TSO6, VIRTIO_NET_F_GUEST_TSO6),
+        ];
+
+        let mut tap_features: u32 = 0;
+        let mut ack_features: u32 = 0;
+
+        for i in 0..desired_features.len() {
+            tap_features |= desired_features[i].0;
+            ack_features |= 1 << desired_features[i].1;
+        }
```

Comment:
And here you also can use `for (tap_feature, virtio_feature) in desired_features.iter()`